### PR TITLE
enhancement: handle large token supplies and metadata

### LINKED ISSUE #4104: [Task]: Handle large values for token supply and metadata
### Task description

TBD

### Requirements

TBD

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary
The protocol allows arbitrary large supplies and decimals for native tokens. If we have such a big value in an asset or activity, a lot of arithmetic bugs occur. 
This PR solves this issue by displaying those numbers in scientific notation (e.g. 4e+20),  

## Changelog

```
- Fixing assets and activities with a very large amount of tokens
- cap upper limit on how many decimals can be put into a transaction
- add tests for `formatTokenAmountDefault` 
```

## Relevant Issues
Closes #4104 

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/utils/isValidIRC30.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { NativeTokenBalance } from '@iota/wallet/out/types'
+import { IPersistedAsset } from '../interfaces/persisted-asset.interface'
+
+export function isValidIRC30(asset: IPersistedAsset, balance: NativeTokenBalance): boolean {
+    const hasValidDecimals = Number(asset.metadata.decimals) < Number.MAX_SAFE_INTEGER
+    const hasValidTotal = Number(balance.total) < Number.MAX_SAFE_INTEGER
+    const hasValidAvailable = Number(balance.available) < Number.MAX_SAFE_INTEGER
+
+    return hasValidDecimals && hasValidTotal && hasValidAvailable
```
Comment (Tuditi) [inline packages/shared/lib/core/wallet/utils/isValidIRC30.ts:9]: This is a limitation of JS, not of IRC30, so the name is incorrect. Possibly we should look into how to handle larger values by using BigInt or Big library in Javascript. Firefly should be able to handle these values.

**Review by Tuditi — CHANGES_REQUESTED** [review]
Code wise one minor thing. I'm not happy with the fact that we have these limitations. Did we consider using the big.js library?

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
This doesn't cover the filters that we have in the activities

**Review by Tuditi — CHANGES_REQUESTED** [review]
Codewise it looks good! As discussed in our call, there are still two things that should be addressed:
1) Let's limit the they can input to 24
2) I experienced a bug when setting the activity filter first to `less than 1` and then updating it to `greater than 1`. Let me know if you need the stronghold backup of my profile to reproduce!

**Review by Tuditi — APPROVED** [review]
LTGM! :fire:

**Pull request merged**